domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsExactlyEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,298 @@
+package org.oppia.android.domain.classify.rules.fractioninput
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.Fraction
+import org.oppia.android.app.model.InteractionObject
+import org.oppia.android.domain.classify.RuleClassifier
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [FractionInputIsExactlyEqualToRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class FractionInputIsExactlyEqualToRuleClassifierProviderTest {
+  /**
+   * Store fractions in format:
+   *    <0 or 1 indicating whether it's negative>, <numerator>, <denominator>
+   *    e.g. -1/2 would be {1, 1, 2}
+   * Store whole numbers in format:
+   *    <0 or 1 indicating whether it's negative>, <whole number>
+   *    e.g. 1234 would be {0, 1234}
+   * Store mixed numbers in format:
+   *    <0 or 1 indicating whether it's negative>, <whole number>, <numerator>, <denominator>
+   *    e.g. -1234 1/2 would be {1, 1234, 1, 2}
+   */
+  private val NON_NEGATIVE_VALUE_0 = createNonNegativeInt(value = 0)
+  private val WHOLE_POS_1 = listOf(0, 123)
```

Comment:
Optional here & for the other constants, but it sometimes helps improve test clarity to include specifics on the value that the test constant encapsulates. In this case, it might be nice to have ``WHOLE_NUMBER_123``, ``WHOLE_NUMBER_321``, ``MIXED_NUMBER_NEGATIVE_123_1_OVER_2`` or similar since then you can clearly see in the test that the values are different, and thus should behave differently.